MagicalRecord/Categories/NSManagedObject/NSManagedObject+MagicalRecord.m

```diff
@@ -287,8 +287,8 @@ - (id) MR_inContext:(NSManagedObjectContext *)otherContext
     
     if ([[self objectID] isTemporaryID])
     {
-        BOOL success = [[self managedObjectContext] obtainPermanentIDsForObjects:@[self] error:&error];
-        if (!success)
+        [[self managedObjectContext] obtainPermanentIDsForObjects:@[self] error:&error];
```

Comment:
I'm a bit worried that if the call to `obtainPermanentIDsForObjects:error:` returns `NO`, this change will ensure that we don't return `nil` here.  That's a significant change in logic.

`+[MagicalRecord handleErrors:]` is safe to call if the `error` parameter is `nil`, so there's no need for it to be wrapped in a check for the existence of the `error` instance.